content/300-guides/050-database/870-using-prisma-with-mongodb.mdx

```diff
@@ -0,0 +1,344 @@
+
+### How to add in missing relations after Introspection
+
+After introspecting an existing MongoDB database, you will need to manually add in relations between models. MongoDB does not have the concept of defining relations via foreign keys, as you would in a relational database. However, if you have a collection in MongoDB with a "foreign-key-like" field that matches the ID field of another collection, Prisma will allow you to emulate relations between the collections.
+
+As an example, take a MongoDB database with two collections, `User` and `Post`. The data in these collections has the following format, with a `userId` field linking users to posts:
+
+`User` collection:
+
+- `_id` field with a type of `objectId`
+- `email` field with a type of `string`
+
+`Post` collection:
+
+- `_id` field with a type of `objectId`
+- `title` field with a type of `string`
+- `userId` with a type of `objectID`
+
+On introspection with `db pull`, this is pulled in to the Prisma schema file as follows:
+
+```prisma file=prisma/schema.prisma
+model Post {
+  id     String @id @default(auto()) @map("_id") @db.ObjectId
+  title  String
+  userId String @db.ObjectId
+}
+
+model User {
+  id    String @id @default(auto()) @map("_id") @db.ObjectId
+  email String
+}
+```
+
+This is missing the relation between the `User` and `Post` models. To fix this, manually add a `user` field to the `Post` model with a `@relation` attribute using `userId` as the `fields` value, linking it to the `User` model, and a `posts` field to the `User` model as the back relation:
+
+```prisma file=prisma/schema.prisma highlight=5;add|11;add
+model Post {
+  id     String @id @default(auto()) @map("_id") @db.ObjectId
+  title  String
+  userId String @db.ObjectId
+  user   User   @relation(fields: [userId], references: [id])
+}
+
+model User {
+  id    String @id @default(auto()) @map("_id") @db.ObjectId
+  email String
+  posts Post[]
+}
+```
+
+For more information on how to use relations in Prisma, see [our documentation](/concepts/components/prisma-schema/relations).
+
+### How to filter for <inlinecode>null</inlinecode> and undefined fields
+
+To understand how MongoDB distinguishes between `null` and undefined fields, consider the example of a `User` model with an optional `name` field:
+
+```ts
+model User {
+  id    String  @id @default(auto()) @map("_id") @db.ObjectId
+  email String
+  name  String?
+}
+```
+
+First, try creating a record with the `name` field explicitly set to `null`. Prisma will return `name: null` as expected:
+
+<CodeWithResult expanded={true}>
+
+<cmd>
+
+```ts
+const createNull = await prisma.user.create({
```

Comment:
```suggestion
const user = await prisma.user.create({
```

Unless depended on elsewhere, `createNull` doesn't really tell what's being returned here.

Reply:
Same with the `createUndefined` below

Reply:
And findNulls

Reply:
Is this a Prisma convention, or some other convention? I'm happy to do it but need to know why and what the other `find`s and `create`s should be changed to

Reply:
A variable called `createNull` does not contain any information in its name what is in it.

Reply:
Yes but it does contain the more salient information for the purposes of this example, that you're creating something with a `null` value. Otherwise, what do we call them? `user` and `user2`? What about the `find`s?

Reply:
Will revisit after the release.